### PR TITLE
feat: pagination and loader added

### PR DESCRIPTION
**Pull request by shobit1337**
Implemented the following features:
- pagination support on the product listing page.
- loader component when there is something loading

Demo:
<a href="https://www.loom.com/share/7ef78231d1b54b69ba01d2ddd6f0ff16">
    <p>Gamotore - Browse - 21 February 2022 - Watch Video</p>
    
  </a>

### CODE REVIEW

File: css/styles.css [hunk 1]
```diff
@@ -443,6 +443,58 @@ header > nav {
...
+/* Pagination */
+.pagination {
+  display: flex;
+  justify-content: center;
+  gap: var(--space-xs);
+}
+
+.pagination a {
+  color: var(--dark-lighter);
+  /* outline: 1px solid; */
```
Comment (Pratik1005) [inline css/styles.css]: Remove code which you are not using

File: css/styles.css [hunk 2]
```diff
@@ -443,6 +443,58 @@ header > nav {
...
+  background-color: var(--dark-light);
+  color: var(--light-color);
+}
+
+/* Loader */
+.loader {
+  border: 4px solid var(--dark-lightest);
+  border-radius: 50%;
+  border-top: 4px solid var(--dark-light);
+  width: 30px;
```
Comment (Pratik1005) [inline css/styles.css]: Use rems as much as possible

**Review by Pratik1005 — APPROVED** [review]
I have commented some point rest looks good

**Review by xsaurabhsuthar — APPROVED** [review]

File: css/styles.css (hunk 1 above)
Comment (shobit1337) [reply css/styles.css]: Thanks, I've removed the comments.

File: css/styles.css (hunk 2 above)
Comment (shobit1337) [reply css/styles.css]: Thanks, I've updated it.

**Pull request merged**